src/prpy/tsr/tsr.py

```diff
@@ -52,36 +71,323 @@ def __init__(self, T0_w=None, Tw_e=None, Bw=None,
             self.manipindex = manip
         self.bodyandlink = bodyandlink
 
-    def sample(self, vals=None):
-        Bwdims = 0
-        for i in range(6):
-            if self.Bw[i, 0] != self.Bw[i, 1]:
-                Bwdims += 1
-
-        if vals is None:
-            Bwvals = self.Bw[:, 0] + (self.Bw[:, 1] - self.Bw[:, 0]) \
-                * numpy.random.random_sample(6)
-        elif len(vals) == Bwdims:
-            Bwvals = numpy.zeros(6)
-            vals_i = 0
-            for i in range(6):
-                if self.Bw[i, 0] != self.Bw[i, 1]:
-                    Bwvals[i] = vals[vals_i]
-                    vals_i += 1
-                else:
-                    Bwvals[i] = self.Bw[i, 0]
-        elif len(vals) == 6:
-            Bwvals = vals
+    @staticmethod
+    def rot_to_rpy(rot):
+        """
+        Converts a rotation matrix to one valid rpy
+        @param rot 3x3 rotation matrix
+        @return rpy 3x1 rpy
+        """
+        rpy = numpy.zeros(3)
+        if not (abs(abs(rot[2, 0]) - 1) < EPSILON):
+            p = -numpy.arcsin(rot[2, 0])
+            rpy[0] = numpy.arctan2((rot[2, 1]/numpy.cos(p)),
+                                   (rot[2, 2]/numpy.cos(p)))
+            rpy[1] = p
+            rpy[2] = numpy.arctan2((rot[1, 0]/numpy.cos(p)),
+                                   (rot[0, 0]/numpy.cos(p)))
+        else:
+            if abs(rot[2, 0] + 1) < EPSILON:
+                r_offset = numpy.arctan2(rot[0, 1], rot[0, 2])
+                rpy[0] = r_offset
+                rpy[1] = pi/2
+                rpy[2] = 0.
+            else:
+                r_offset = numpy.arctan2(-rot[0, 1], -rot[0, 2])
+                rpy[0] = r_offset
+                rpy[1] = -pi/2
+                rpy[2] = 0.
+        return rpy
+
+    @staticmethod
+    def trans_to_xyzrpy(trans):
+        """
+        Converts a transformation matrix to one valid xyzrpy
+        @param trans 4x4 transformation matrix
+        @return xyzrpy 6x1 xyzrpy
+        """
+        xyz, rot = trans[0:3, 3], trans[0:3, 0:3]
+        rpy = TSR.rot_to_rpy(rot)
```

Comment:
I'd prefer to make this a `@classmethod` and replace `TSR.rot_to_rpy` with `cls.rot_to_rpy`. It's nice to DRY if we ever have to rename the class.

We'll probably never rename `TSR`, so this is just a nitpick. Feel free to disregard.